cg/services/validate_file_transfer_service/validate_file_transfer_service.py

```diff
@@ -5,34 +5,49 @@
 
 
 class ValidateFileTransferService:
-    """Service to validate file transfers."""
+    """Service to validate file transfers via a manifest file."""
+
+    def validate_file_transfer(
+        self, manifest_file: Path, source_dir: Path, manifest_file_format: str
+    ):
+        """Validate all files listed in the manifest are present in the directory tree."""
+        files_in_manifest: list[str] = self._get_files_in_manifest(
+            manifest_file=manifest_file, manifest_file_format=manifest_file_format
+        )
+        if not self._are_all_files_present(
+            files_to_validate=files_in_manifest, source_dir=source_dir
+        ):
+            raise FileNotFoundError(
+                f"Not all files listed in the manifest file {manifest_file} are not present in the directory tree {source_dir}"
+            )
+        return
 
     @staticmethod
-    def get_manifest_file_content(manifest_file: Path, manifest_file_format: str) -> list[str]:
+    def _get_manifest_file_content(manifest_file: Path, manifest_file_format: str) -> list[str]:
         """Get the content of the manifest file."""
         file_reader = ReadFile()
         return file_reader.get_content_from_file(
             file_format=manifest_file_format, file_path=manifest_file
         )
 
     @staticmethod
-    def is_valid_path(line: str):
+    def _is_valid_path(line: str):
         return "/" in line
```

Comment:
Out of scope probably but this method seems off to me. Is there not a better check to do?

Reply:
Like a Path.is_dir()?

Reply:
Yes I agree, I will have a look at this in a different pr.